### ISSUE TITLE
Support Traversable object in querybuilder.

### ISSUE THREAD

**Issue by SamMousa** [yiisoft/yii2#10954]
It would be nice if `QueryBuilder` supported `Traversable` objects.
This works:
````
$query = new Query();
$userIds = [1, 5, 6];
$query->where(['id' => $userIds]);
````

This doesn't:
````
$query = new Query();
$userIds = new ArrayObject([1, 5, 6]);
$query->where(['id' => $userIds]);
````

The changes required to support this would be minor and would not break BC.

-- Related to this is support for `\Traversable` objects in `ExistValidator`

**Comment by SamMousa** [comment]
Will implement this when #10953 is merged.

**Comment by SilverFire** [comment]
#10953 is merged

**Comment by cebe** [comment]
:+1:

**Comment by SamMousa** [comment]
Bump

**Comment by cebe** [comment]
@SamMousa your PR is merged in 2.0.8, so this can be closed, right?

**Comment by cebe** [comment]
https://github.com/yiisoft/yii2/commit/baab8d6a68e46394c04125eef094003527ab08e1
